File: java/src/museo/MonitorVestibulo.java
package museo;

import java.util.Collections;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class MonitorVestibulo {

    private final int idMonitor;
    private final String ubicacion;
    private final Map<Integer, Sala> salas = new HashMap<>();

    public MonitorVestibulo(int idMonitor, String ubicacion, List<Sala> listaSalas) {
        this.idMonitor = idMonitor;
        this.ubicacion = ubicacion;
        for (Sala s : listaSalas) {
            salas.put(s.getIdSala(), s);
        }
    }

    public int getIdMonitor() {
        return idMonitor;
    }

    public String getUbicacion() {
        return ubicacion;
    }

    public List<ObraArte> mostrarListadoPorSala(int idSala) {
        Sala sala = salas.get(idSala);
        if (sala == null) {
            return Collections.emptyList();
        }
        return sala.listarObras();
    }
}

File: javascript/src/museo/monitorVestibulo.js
export class MonitorVestibulo {
  /**
   * @param {number} idMonitor
   * @param {string} ubicacion
   * @param {import('./sala.js').Sala[]} listaSalas
   */
  constructor(idMonitor, ubicacion, listaSalas) {
    this.idMonitor = idMonitor;
    this.ubicacion = ubicacion;
    /** @type {Map<number, import('./sala.js').Sala>} */
    this.salas = new Map();
    for (const s of listaSalas) {
      this.salas.set(s.getIdSala(), s);
    }
  }

  getIdMonitor() {
    return this.idMonitor;
  }

  getUbicacion() {
    return this.ubicacion;
  }

  mostrarListadoPorSala(idSala) {
    const sala = this.salas.get(idSala);
    if (sala == null) {
      return [];
    }
    return sala.listarObras();
  }
}
